snapcraft/pluginhandler.py

```diff
@@ -268,17 +278,20 @@ def build(self, force=False):
         self.code.build()
         self.mark_done('build')
 
-    def clean_build(self):
+    def clean_build(self, hint=''):
         state_file = self._step_state_file('build')
         if not os.path.isfile(state_file):
             self.notify_part_progress('Skipping cleaning build for',
                                       '(already clean)')
             return
 
-        self.notify_part_progress('Cleaning build for')
+        self.notify_part_progress('Cleaning build for', hint)
         self.code.clean_build()
         self.mark_cleaned('build')
 
+    def fileset_for(self, step):
+        return getattr(self.code.options, step, ['*']) or ['*']
```

Comment:
@elopio good catch, this should probably replace
https://github.com/ubuntu-core/snapcraft/pull/442/files#diff-2185a0b0812f4f8b12703640a5d79517R297